pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -191,19 +198,84 @@ func evictPodsFromTargetNodes(client clientset.Interface, evictionPolicyGroupVer
+				// TODO: This is O(n^2), we can better this.
+				groupByPodPriority(*pod.Spec.Priority, node.allpods, podPriorityGroup)
+			}
+
+			// evict pods based on priority.
+			for _, pods := range podPriorityGroup {
+				// Pods within same group needs to evicted based on QoS.
+				nonEvictablePods, bePods, bPods, gPods := classifyPods(pods)
+				glog.V(1).Infof("There are %v non-evictable pods on the node", len(nonEvictablePods))
+				evictPods(bePods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
+				evictPods(bPods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
+				evictPods(gPods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
+			}
+		} else {
+			// TODO: Remove this when we support only priority.
+			//  Falling back to evicting pods based on priority.
+			glog.V(1).Infof("Evicting pods based on QoS")
+			glog.V(1).Infof("There are %v non-evictable pods on the node", len(node.nonRemovablePods))
+			// evict best effort pods
+			evictPods(node.bePods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
+			// evict burstable pods
+			evictPods(node.bPods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
+			// evict guaranteed pods
+			evictPods(node.gPods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
+		}
 		nodepodCount[node.node] = currentPodsEvicted
 		podsEvicted = podsEvicted + nodepodCount[node.node]
 		glog.V(1).Infof("%v pods evicted from node %#v with usage %v", nodepodCount[node.node], node.node.Name, node.usage)
 	}
 	return podsEvicted
 }
 
+// groupByPodPriority groups pods by priorities.
+func groupByPodPriority(priority int32, pods []*v1.Pod, podGroupByPriority priorityToPodMap) {
+	podsWithSamePriority := []*v1.Pod{}
+	for _, pod := range pods {
+		if *pod.Spec.Priority == priority {
+			if _, ok := podGroupByPriority[priority]; !ok {
+				podGroupByPriority[priority] = append(podsWithSamePriority, pod)
+			} else {
+				// add only if the pod doesn't exist in current list.
+				if !checkIfPodExistsInList(pod, podGroupByPriority[priority]) {
+					podGroupByPriority[priority] = append(podGroupByPriority[priority], pod)
+				}
+			}
+		}
+	}
+}
+
+// checkIfAllPodsHavePriority checks if all the pods in the given pod list have a priority associated with them.
+func checkIfAllPodsHavePriority(podList []*v1.Pod) bool {
+	for _, pod := range podList {
```

Comment:
why are you checking it for all pods? if one pod has priority, that means all have priority, if one does not have, that means, all dont have priority, so just checking for one pod should be enough?

Reply:
Well, I am not sure, if priority admission plugin is enabled after some pods are created, there could be some pods that don't have priority.(Because admission controller is run on create/update). While I did not test this scenario, I think this could be a problem, that's why I ran for entire set of pods on the node. This also ensures that the subsequent sorting of pods based on priority won't fail.